src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "jitpch.h"
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+#ifdef FEATURE_HW_INTRINSICS
+
+#include "emit.h"
+#include "codegen.h"
+#include "sideeffects.h"
+#include "lower.h"
+#include "gcinfo.h"
+#include "gcinfoencoder.h"
+
+//------------------------------------------------------------------------
+// genIsTableDrivenHWIntrinsic:
+//
+// Arguments:
+//    category - category of a HW intrinsic
+//
+// Return Value:
+//    returns true if this category can be table-driven in CodeGen
+//
+static bool genIsTableDrivenHWIntrinsic(NamedIntrinsic intrinsicId, HWIntrinsicCategory category)
+{
+    // TODO - make more categories to the table-driven framework
+    const bool tableDrivenCategory = (category != HW_Category_Scalar) && (category != HW_Category_Helper);
+    const bool tableDrivenFlag = true;
+    return tableDrivenCategory && tableDrivenFlag;
+}
+
+//------------------------------------------------------------------------
+// genHWIntrinsic: Generates the code for a given hardware intrinsic node.
+//
+// Arguments:
+//    node - The hardware intrinsic node
+//
+void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
```

Comment:
A lot of this logic is actually very similar to the x86 logic, there might be more opportunities to share bits here as well, but that isn't part of this PR.